src/library/scala/collection/Iterator.scala

```diff
@@ -1259,7 +1259,10 @@ object Iterator extends IterableFactory[Iterator] {
         else adjustedBound min (until - lo)   // keep lesser bound
       if (rest == 0) empty
       else {
-        dropping += lo
+        dropping = {
+          val sum = dropping + lo
+          if (sum < 0) Int.MaxValue else sum
+        }
```

Comment:
the `sum` is not a good name,maybe you can find another one.

Reply:
Its scoped down to dropping only which makes the choice of name quite unnecessary